Source/WebKit/UIProcess/ProvisionalFrameProxy.h

```diff
@@ -68,7 +68,7 @@ class ProvisionalFrameProxy : public IPC::MessageReceiver, public IPC::MessageSe
     IPC::Connection* messageSenderConnection() const final;
     uint64_t messageSenderDestinationID() const final;
 
-    WebFrameProxy& m_frame;
+    WeakPtr<WebFrameProxy> m_frame;
```

Comment:
I assume this cannot be null. If so, using a WeakPtr is awkward because it makes it look like it could be null. Why not use a CheckedRef?

Reply:
It currently doesn't support a CheckedRef, so I wasn't sure if adding that overhead was worth it for this object. If you think that's ok, I'm happy to add it.

Reply:
Yes, I believe it is OK to add support for CheckedRef.

Reply:
ok, I see the same pattern used in ProvisionalPageProxy, will fix that too